### ISSUE TITLE
Box2D 2.4.0

### ISSUE THREAD

**Issue by LeXXik** [kripken/box2d.js#114]
Hi, all! Thank you for the effort!
Are there any plans on adding support to the new Box2D version 2.4.0?

**Comment by Birch-san** [comment]
I've forked `box2d.js` and added support for Box2D v2.4.0:  
﻿https://github.com/Birch-san/box2d-wasm

**Comment by kripken** [comment]
I don't have time to maintain this project myself, but I'd be happy to merge pull requests like that, @Birch-san

**Comment by Birch-san** [comment]
thanks! :) I'm not sure what that pull request would look like — box2d-wasm isn't a fork in the strictest sense — it's a "start again" but with copying + modifying of a few files (the .idl bindings, glue stub, helpers), and with an entirely different approach to build and source control.

I think it'd be relatively easy to contribute TypeScript typings to box2d.js (i.e. by running https://github.com/Birch-san/box2d-wasm/tree/master/webidl-to-ts on each .idl file you have), or to contribute a .wasm and .idl of the Box2D v2.4.0+@[f0f9d50](https://github.com/erincatto/box2d/tree/f0f9d50a328a709cc3a287a61b864e7d0e3ef35f) commit that I'm on, but anything beyond that would be substantially harder (e.g. due to the cmake-based build system being very different to the Makefile-based build system).

**Comment by joex92** [comment]
THanks @Birch-san, What about Box2D 2.4.1? Does 2.4.0 fixes issue #90 ?

**Comment by Birch-san** [comment]
@JxM-92 I don't know whether the small block allocator has been made any less leaky in newer versions, but the repro looks simple so should be easy to check.

I did actually try upgrading to Box2D v2.4.1, but it instantly encounters assertion failures at runtime (fails to assert that inertia must be positive). this is something that needs reporting as an issue. I briefly looked at the commits that had happened in the intervening time, and wonder whether it's related to reverting back to an older interpolation algorithm.

**Comment by kripken** [comment]
@Birch-san In that case, maybe a link in this repo could go to yours?

**Comment by Birch-san** [comment]
@kripken that sounds good, if you'd be okay with that?

**Comment by kripken** [comment]
@Birch-san Yeah, it sounds like a good idea to link to it from the readme, could help people find it.

If people find it useful maybe we can merge the repos, that's another option. But for now a link is a good simple step.
